### PR TITLE
Gnumeric

### PR DESCRIPTION
**Pull request by cedlemo**

### CODE REVIEW

**Comment by cosmo0920** [comment]
Could you add `gir1.2-gnumeric` package into [travis-before-script.sh](https://github.com/ruby-gnome2/ruby-gnome2/blob/master/travis-before-script.sh) like this commit: https://github.com/ruby-gnome2/ruby-gnome2/commit/3529b12440b71be1fce8f954b6c3eff6b938f357?
It would be more helpful for us.

**Comment by cedlemo** [comment]
Done.

File: gnm/test/run-test.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+modules.each do |target, module_name|
+  makefile = File.join(target, "Makefile")
+  if File.exist?(makefile) and system("which make > /dev/null")
+    `make -C #{target.dump} > /dev/null` or exit(false)
+  end
+  $LOAD_PATH.unshift(File.join(target, "ext", module_name))
+  $LOAD_PATH.unshift(File.join(target, "lib"))
+end
+
+$LOAD_PATH.unshift(File.join(gsf_base, "test"))
```
Comment (cosmo0920) [inline gnm/test/run-test.rb]:
```diff
-gsf_base
+gnm_base
```

File: gnm/test/run-test.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+ruby_gnome2_base = File.join(File.dirname(__FILE__), "..", "..")
+ruby_gnome2_base = File.expand_path(ruby_gnome2_base)
+
+gobject_introspection_base = File.join(ruby_gnome2_base, "gobject-introspection")
+goffice_base = File.join(ruby_gnome2_base, "goffice")
```
Comment (cosmo0920) [inline gnm/test/run-test.rb]: It seems to be add the following line:

```ruby
gnm_base = File.join(ruby_gnome2_base, "gnm")
```

**Comment by cedlemo** [comment]
@cosmo0920,

Oups, It looks like I was tired when I did that yesterday. Hope there are no other mistakes.

File: gnm/test/run-test.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+gnm_base = File.join(ruby_gnome2_base, "gnm")
+
+modules = [
+  [gobject_introspection_base, "gobject-introspection"],
+  [goffice_base, "goffice"]
+
```
Comment (cosmo0920) [inline gnm/test/run-test.rb]: OMG! goffice also depends gdk3! https://travis-ci.org/ruby-gnome2/ruby-gnome2/jobs/132219186#L2287

**Comment by cosmo0920** [comment]
A good work depends on a good quality of sleep.
Please have a nice rest tonight. ;)

**Comment by cedlemo** [comment]
> OMG! goffice also depends gdk3

Yes but I though that this will be solved with the dependency of Gtk3.

You know GOffice load Gtk3, which load Gdk3 and Glib and Gio. Did I misunderstand that?

**Comment by cosmo0920** [comment]
> You know GOffice load Gtk3, which load Gdk3 and Glib and Gio. Did I misunderstand that?

Unfortunately, yes.
`GOffice` loads `Gtk3` at runtime not testing time.
When we test a gem on Ruby-GNOME2, we should specify its dependent gems explicitly.

**Comment by cedlemo** [comment]
So I guess I since Goffice depends on Gtk3 like Vte3, I should use almost the same dependencies. I will do this for Gnumeric too.  Here is what I plan to had for the Goffice and Gnumeric:

```
glib_base = File.join(ruby_gnome2_base, "glib2")
atk_base = File.join(ruby_gnome2_base, "atk")
pango_base = File.join(ruby_gnome2_base, "pango")
gdk_pixbuf_base = File.join(ruby_gnome2_base, "gdk_pixbuf2")
cairo_gobject_base = File.join(ruby_gnome2_base, "cairo-gobject")
gobject_introspection_base = File.join(ruby_gnome2_base, "gobject-introspection")
gio2_base = File.join(ruby_gnome2_base, "gio2")
gdk3_base = File.join(ruby_gnome2_base, "gdk3")
gtk3_base = File.join(ruby_gnome2_base, "gtk3")
...

modules = [
  [glib_base, "glib2"],
  [atk_base, "atk"],
  [pango_base, "pango"],
  [cairo_gobject_base, "cairo-gobject"],
  [gdk_pixbuf_base, "gdk_pixbuf2"],
  [gobject_introspection_base, "gobject-introspection"],
  [gio2_base, "gio2"],
  [gdk3_base, "gdk3"],
  [gtk3_base, "gtk3"],
 ...
]
```

I will way for an answer before doing it.

**Comment by cedlemo** [comment]
@cosmo0920 , @kou ,

I have added the needed code in order to solve the dependencies in the test file. Now I get an error https://travis-ci.org/ruby-gnome2/ruby-gnome2/jobs/132494966#L1618 which is more related to the naming /renaming of the GOffice classes. When I test it on my system, everything is fine.

Notheless I think that this PR should be merged and that I should check the error after that.

File: gnm/test/test-input-stdio.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# Copyright (C) 2016  Ruby-GNOME2 Project Team
```
Comment (kou) [inline gnm/test/test-input-stdio.rb]: This file is needless. It's a test for GSF.

**Comment by kou** [comment]
I prefer to use "Gnumeric" instead of "Gnm" for library name like "GStreamer".

**Comment by cedlemo** [comment]
The story of this pull request : 

![this PR]
(http://i.imgur.com/WXEYfV4.gif)

**Comment by kou** [comment]
Thanks!

**Comment by cosmo0920** [comment]
Awesome!

**Pull request merged**